### ISSUE TITLE
Program check not detected when PER trace active

### ISSUE THREAD

**Issue by arfineman** [SDL-Hercules-390/hyperion#560]
HHC01413I Hercules version **4.5.0.10830-SDL-g58578601** &nbsp;_(i.e. official 4.5 release)_

This issue appears to be related to issues #222 and #295. It is of low importance, as it only fails when PER trace is active. 

I was able to simplify it to a few instructions.

Here is a sample program:
```
BAD      START ,
         USING BAD,12
         ST    14,CMSR14
         DC    X'00000000'
         DC    X'00000000'    <----   Add as many as desired
         DC    X'00000000'
         DC    X'00000000'
         SR    15,15
         L     14,CMSR14
         BR    14
CMSR14   DC    F'0'
         END   ,
```
Here is the results when trace is not active:
```
 load bad ( start
DMSLIO740I Execution begins...
DMSABE141T Operation exception occurred at 00020006 in routine BAD
CMS
```

Here is the results with trace active:
```
 tr inst r 20000.1ff run
Ready; T=0.02/0.04
 load bad ( start
DMSLIO740I Execution begins...
 -> 00020000  ST      50E0C01C >> 0002001C    CC 1
 -> 00020014  SR      1BFF        CC 0
    00020016  L       58E0C01C    0002001C    CC 0
    0002001A  BR      07FE     -> 00F0EAC4    CC 0
Ready; T=0.01/0.01
```

**Comment by Fish-Git** [comment]
What of VM are you running? VM/370? z/VM? Which version?

**Comment by arfineman** [comment]
Sorry, I should have mentioned that. It is VM/ESA 2.4. Probably reproducible on z/VM. 
It works correctly on 4.2.1. That's why I suspect it is related to issue #222 and #295. 
Best regards, 

HHC01413I Hercules version 4.2.1.9826-SDL-g9b5cbe3d (4.2.1.9826)
 
CP TR INST R 20000.1FF RUN                                            
 load bad ( start                                                      
DMSLIO740I Execution begins...                                         
 -> 00020000  ST      50E0C01C >> 0002001C    CC 1                     
    00020004  ????    0000                                             
*** 00020004      PROG    0001 -> 0139B5C0        OPERATION            
DMSABE141T Operation exception occurred at 00020006 in routine BAD     
CMS

**Comment by Fish-Git** [comment]
> Probably reproducible on z/VM.

Confirmed. Same behavior with z/VM 7.3.

Researching...

**Comment by arfineman** [comment]
Hi Fish,
This test case and the one for issue #295 work correctly on 4.2.0. It appears the performance fix for issue #222 broke this.
Best regards,

**Comment by Fish-Git** [comment]
> It appears the performance fix for issue #222 broke this.

Nope. What broke things was a careless typo on my part introduced by commit 2dc33fe76d9a2126935fde63c71b5ad1986eb0e6.

The following patch fixes it:

```diff
```

Fixed by commit 376d445f00ae0beb4a96a3c4cde41cf399589daf.

Closing.
